src/sherpa_ai/tools.py

```diff
@@ -202,3 +211,43 @@ def _run(self, query: str) -> str:
 
     def _arun(self, query: str) -> str:
         raise NotImplementedError("UserInputTool does not support async run")
+
+class HugChatTool(BaseTool):
+    name = "Hugchat"
+    description = (
+        "Access the user input for the task."
+        "You use this tool if you need to use HugChat for Q&A."
+    )
+    def _run(self, query: str) -> str:
+
+        email = cfg.HUGCHAT_EMAIL
+        passwd = cfg.HUGCHAT_PASS
+        # Log in to huggingface and grant authorization to huggingchat
+        sign = Login(email, passwd)
+        cookies = sign.login()
+
+        # Save cookies to the local directory
+        cookie_path_dir = "./cookies_snapshot"
+        sign.saveCookiesToDir(cookie_path_dir)
+
+        # Create a ChatBot
+        chatbot = hugchat.ChatBot(cookies=cookies.get_dict())  # or cookie_path="usercookies/<email>.json"
+
+        query_result = ""
+
+        if cfg.HUGCHAT_CONFIG == 1:
+           query_result = chatbot.query(query)
+
+
+        if cfg.HUGCHAT_CONFIG == 2:
```

Comment:
The integer values for `cfg.HUGCHAT_CONFIG` are not self-explanatory. This means users of this module will have to read the code in order to understand how each configuration value works. 

Here are a few alternatives approaches that are more readable:

1. In `src/sherpa_ai/config.py` define constants for the different modes of operation. For example, 
```
HUGCHAT_MODE_SIMPLE_QUERY = 1     # Simple query with no extra options
HUGCHAT_MODE_STREAM_RESPONSE = 2  # Use the stream response option
...
```
and use these constants as comparison values in `tools.py`

2. If the modes aren't mutually exclusive then define config values to turn each mode on and off, e.g.
```
HUGCHAT_STREAM = environ.get("HUGCHAT_STREAM", False) == "True"
HUGCHAT_WEB_SEARCH = environ.get("HUGCHAT_WEB_SEARCH", False) == "True"
```
 and, in tools.py, something like `chatbot.query(query, stream=HUGCHAT_STREAM, web_search=HUGCHAT_WEB_SEARCH, ...)`